src/app/features/filter-bar/containers/filter-bar/filter-bar.component.ts

```diff
@@ -1,127 +1,110 @@
-    {
-      id: 'Tel Aviv',
-      isChecked: false,
-      title: 'Tel Aviv',
-    },
-    {
-      id: 'New York',
-      isChecked: false,
-      title: 'New York',
-    },
-    {
-      id: 'Barcelona',
-      isChecked: false,
-      title: 'Barcelona',
-    },
-  ];
+  @Input() products: Array<Product>;
 
   categories$: Observable<Option[]> = combineLatest([
-      of(this.categories),
+      this.categoryService.getCategories(),
       this.filteringStore.select(getCategoryFilterValue)
     ]
   ).pipe(
     map(([categories, selectedCategories]) => {
-      categories.forEach((category) => {
+      let categoriesOptions: Array<Option> = categories.map((category: Category) => {
+        return {
+          id: category.id,
+          title: category.name,
+          amount: this.products.filter((product) => product.categoryId === category.id).length
+        } as Option;
+      });
+      categoriesOptions.forEach((category) => {
         if (selectedCategories) {
           category.isChecked = selectedCategories.includes(category.id);
         } else {
           category.isChecked = false;
         }
       });
 
-      return categories;
+      return categoriesOptions;
     })
   );
 
-  cities$: Observable<Option[]> = combineLatest(
-    of(this.cities),
-    this.filteringStore.select(getCityFilterValue)
+  countries$: Observable<Option[]> = combineLatest([
+      this.addressesService.getCountries(),
+      this.filteringStore.select(getCountryFilterValue)
+    ]
   ).pipe(
-    map(([cities, selectedCities]) => {
-      cities.forEach((city) => {
+    map(([countries, selectedCities]) => {
+      let countriesOptions: Array<Option> = countries.map((country) => {
+        return {
+          id: country,
+          title: country,
+          amount: Math.floor(Math.random() * 301) // TODO: Need to add the correct logic
+        } as Option;
+      });
+      countriesOptions.forEach((city) => {
         if (selectedCities) {
           city.isChecked = selectedCities.includes(city.id);
         } else {
           city.isChecked = false;
         }
       });
 
-      return cities;
+      return countriesOptions;
     })
   );
 
-  constructor(private filteringStore: Store<FilteringState>) {
+  cities$ = this.filteringStore.select(getCountryFilterValue).pipe(
+    switchMap((selectedCountries) => {
+      if (selectedCountries) {
+        return combineLatest([
+          this.addressesService.getCitiesByCountries(selectedCountries),
+          this.filteringStore.select(getCityFilterValue)
+        ]).pipe(
+          map(([cities, selectedCities]) => {
+            let citiesOptions: Array<Option> = cities.map((city) => {
+              return {
+                id: city,
+                title: city,
+                amount: Math.floor(Math.random() * 201) // TODO: Need to add the correct logic
+              } as Option;
+            });
+            citiesOptions.forEach((city) => {
+              if (selectedCities) {
```

Comment:
It's very similar to the countries filter code, can we reuse something?

Reply:
I think so, can you find me proper name for that function?

Reply:
Maybe later, as long as it works it's good